### ISSUE TITLE
[Request for help/explanation] How should strapi actually get deployed to production?

### ISSUE THREAD

**Issue by kubal5003** [strapi/strapi-docker#143]
I found Strapi deployment to be puzzling at least. 

I started with official Docker image and tried deploying it. Everything is fine for a demo run, but then this image is simply not useful or I have no idea how to make use of it. Why is that image even there if all the useful code is then generated in the mounted filesystem?

After generating my Strapi application I figured that I should prepare a Docker image out of it - and that would be all fine (you could even add a default Dockerfile to the app out of the box) except the fact that when I did some changes i.e. added data types - preparing a new image is not enough, because database changes are also required. 

TL; DR;
Can anyone explain the deployment workflow here? I.e. I have a already deployed Strap App and I want to add another data type - what do I do?

**Comment by WillsonHaw** [comment]
> Why is that image even there if all the useful code is then generated in the mounted filesystem?

Make sure you have the correct folder specified in the volume when running your docker container:

```
    environment:
      - APP_NAME=your_app
    volumes:
      - "./api:/usr/src/api/your_app"
```

If set up correctly, the generated code should now be in your `./api` folder, which you can commit.

**Comment by msreekm** [comment]
Agree ,documentation is so confusing on deployment to production.

**Comment by roycornelissen** [comment]
@WillsonHaw @kubal5003 I guess we're looking for the other way around... Let's say you have created those files and put them into a Git repo. Now I want to deploy that to a cloud provider... how to create an image from that?

@kubal5003 did you get any further with that? I'm just as confused. I really like Strapi as a concept and local development works fine. Only the confusing documentation on how to get it through a workflow and put it live is really putting me off. Makes me consider using a different CMS.

The `strapi/strapi` image on Docker Hub is also hopelessly behind. It's an alpha build, and looking at the update history, there isn't a steady flow of updates, so the Docker image seems a total afterthought.

**Comment by alexandrebodin** [comment]
HI, 

We just released a full rebuild of our docker image 🎉 
We are closing the issues and open PRs to start fresh.

If you feel like something is missing or want to add things feel free to open new PRs ! 💯
